presto-hive/src/main/java/com/facebook/presto/hive/metastore/glue/GlueHiveMetastore.java

```diff
@@ -0,0 +1,734 @@
+        AWSGlueAsyncClientBuilder asyncGlueClientBuilder = AWSGlueAsyncClientBuilder.standard()
+                .withClientConfiguration(clientConfig);
+
+        if (config.getGlueRegion().isPresent()) {
+            asyncGlueClientBuilder.setRegion(config.getGlueRegion().get());
+        }
+        else if (config.getPinGlueClientToCurrentRegion()) {
+            Region currentRegion = Regions.getCurrentRegion();
+            if (currentRegion != null) {
+                asyncGlueClientBuilder.setRegion(currentRegion.getName());
+            }
+        }
+
+        return asyncGlueClientBuilder.build();
+    }
+
+    @Override
+    public Optional<Database> getDatabase(String databaseName)
+    {
+        try {
+            GetDatabaseResult result = glueClient.getDatabase(new GetDatabaseRequest().withName(databaseName));
+            return Optional.of(GlueToPrestoConverter.convertDatabase(result.getDatabase()));
+        }
+        catch (EntityNotFoundException e) {
+            return Optional.empty();
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public List<String> getAllDatabases()
+    {
+        try {
+            List<String> databaseNames = new ArrayList<>();
+            String nextToken = null;
+
+            do {
+                GetDatabasesResult result = glueClient.getDatabases(new GetDatabasesRequest().withNextToken(nextToken));
+                nextToken = result.getNextToken();
+                result.getDatabaseList().stream().forEach(database -> databaseNames.add(database.getName()));
+            }
+            while (nextToken != null);
+
+            return databaseNames;
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    @Override
+    public Optional<Table> getTable(String databaseName, String tableName)
+    {
+        try {
+            GetTableResult result = glueClient.getTable(new GetTableRequest()
+                    .withDatabaseName(databaseName)
+                    .withName(tableName));
+            return Optional.of(GlueToPrestoConverter.convertTable(result.getTable(), databaseName));
+        }
+        catch (EntityNotFoundException e) {
+            return Optional.empty();
+        }
+        catch (AmazonServiceException e) {
+            throw new PrestoException(HIVE_METASTORE_ERROR, e);
+        }
+    }
+
+    private Table getTableOrElseThrow(String databaseName, String tableName)
+    {
+        return getTable(databaseName, tableName)
+                .orElseThrow(() -> new TableNotFoundException(new SchemaTableName(databaseName, tableName)));
+    }
+
+    @Override
+    public Optional<Map<String, HiveColumnStatistics>> getTableColumnStatistics(String databaseName, String tableName, Set<String> columnNames)
+    {
+        return Optional.of(ImmutableMap.of());
```

Comment:
is that because Glue does not support statistics?

Reply:
That is correct, though they may add support for it in the future.